chain/client/src/client.rs

```diff
@@ -1609,7 +1609,17 @@ impl Client {
             BlockStatus::Next => {
                 // If this block immediately follows the current tip, remove
                 // transactions from the tx pool.
-                self.remove_transactions_for_block(validator_id, block).unwrap_or_default();
+                match self.remove_transactions_for_block(&validator_id, block) {
+                    Ok(()) => (),
+                    Err(err) => {
+                        tracing::warn!(
+                            "validator {}: removing txs for block {:?} failed with {:?}",
+                            validator_id,
+                            block,
+                            err
+                        );
```

Comment:
As always the rule of thumb for warnings is: can operators do anything about this event? If not, it should not be a warning, but rather a ≤ debug. Otherwise we'll be seeing operators complaining to our support people about weird warnings in their logs and asking if they need to do anything about this.

Alternatively is this event severe enough that we want to have node operators come and tell us about them?

Reply:
to be honest, I do not know how severe this issue is.  I think that this is an oversight and so I intuitively went with a `warn`.  I'll start a zulip thread.

Reply:
fixed.